source/slang/slang-ir-autodiff-unzip.cpp

```diff
@@ -425,6 +429,9 @@ IRFunc* DiffUnzipPass::extractPrimalFunc(
                             inst->getFullType(),
                             intermediateVar,
                             structKeyDecor->getStructKey());
+
+                        val->sourceLoc = user->sourceLoc;
```

Comment:
Instead of setting source locs everywhere, consider use `IRBuilderSourceLocRAII sourceLocInfo(getBuilder(), loc);` to simplify the code. This is an RAII object that will make all future insts emited by the builder automatically have the `loc` as source loc, until the RAII object goes out of scope.